### DISCUSSION TITLE
PHOTOPRISM_SITE_URL and share links in the Docker container

### DISCUSSION THREAD

**Discussion by naasking** [photoprism/photoprism#4623] [Q&A]
I'm running the latest [Docker image](https://registry.hub.docker.com/r/photoprism/photoprism/) and everything seems to be working well now that I tweaked the Sqlite to use the WAL (lots of timeouts while indexing), but I just can't seem to get the site to generate share links with the correct site URL. I've set PHOTOPRISM_SITE_URL env variable as I have all of my other parameters, but all share links still generate with http://localhost:2342/

Any thoughts? A reverse proxy sits in front and handles the https connections, so all traffic to and from the local Docker image is over http. Might that be it?

**Comment by omerdduran** [comment]
Hello, firstly make sure that PHOTOPRISM_SITE_URL is set correctly. PHOTOPRISM_SITE_URL: "https://yourdomain.com/"

URL must contain protocol (http:// or https://)
URL must have a slash (/) at the end
The value must not have a space

Also, since the reverse proxy already handles traffic over HTTPS, the internal network communication between PhotoPrism and the reverse proxy does not need a second SSL layer, so you can set PHOTOPRISM_DISABLE_TLS to "true".

**Comment by naasking** [comment]
I've disabled TLS per your suggestion, thanks. No change on the sharing behaviour though. I did have it configured it correctly per your description:

![image](https://github.com/user-attachments/assets/3a8d0304-e525-4b12-8ebf-351be692f9d9)

**Reply by omerdduran** [reply]
Hi again. Unfortunately, I've seen this problem happen to a few other people, and their solution is to kill the container, edit the setting in the docker compose file, and recreate the container as before. You can try this if you like

**Reply by naasking** [reply]
Do I have to nuke the database as well? If I could preserve the data I'll try that. Alternately, if I can just open the Sqlite file or some other config file manually and edit it, I'm happy to do that, I just need to know where to look.

**Reply by lastzero** [reply]
No. Generally, this only happens if there's a typo in the configuration or the services haven't been properly recreated after performing changes.

**Reply by lastzero** [reply]
As explained in our Getting Started guide, it's also possible that a different url is set in the options.yml file (if one exists). In that case, it will override the corresponding environment variable.

**Reply by naasking** [reply]
I couldn't find an options.yml file in the docker container. I created the file under /photoprism/storage/options.yml with only this setting:

     SiteUrl: "https://photo.[my host name].com/"

Share URLs still come up as "localhost". So I suppose that recreating the container is the only option at this point.

**Reply by lastzero** [reply]
Unless you have changed the configuration, the default mount/storage path for the `options.yml` file is `/photoprism/storage/config/options.yml` - see the following documentation:

- https://docs.photoprism.app/getting-started/config-files/

**Reply by naasking** [reply]
That did it, thanks! Just FYI, I did that read that documentation and this is one of those things that's only obvious in retrospect. The meandering discussion of both config and storage paths muddies the description. I recommend adding a specific example, eg. "For example, if storage path is /photoprism/storage, then you can find it under /photoprism/storage/config/options.yml".
